cmd/crank/beta/top/top.go

```diff
@@ -0,0 +1,260 @@
+
+Examples:
+  # Show resources utilization for all Crossplane pods in the default 'crossplane-system' namespace in a tabular format.
+  crossplane beta top
+
+  # Show resources utilization for all Crossplane pods in a specified namespace in a tabular format.
+  crossplane beta top -n <namespace>
+
+  # Show resources utilization for all Crossplane pods in the default 'crossplane-system' in a split format
+  crossplane beta top -s
+`
+}
+
+type topMetrics struct {
+	PodType      string
+	PodName      string
+	PodNamespace string
+	CPUUsage     int64
+	MemoryUsage  string
+}
+
+type defaultPrinterRow struct {
+	podType   string
+	namespace string
+	name      string
+	cpu       string
+	memory    string
+}
+
+func (r *defaultPrinterRow) String() string {
+	return strings.Join([]string{
+		r.podType,
+		r.namespace,
+		r.name,
+		r.cpu,
+		r.memory,
+	}, "\t") + "\t"
+}
+
+// Run runs the top command.
+func (c *Cmd) Run(k *kong.Context, logger logging.Logger) error { //nolint:gocyclo // TODO:(piotr1215) refactor to use dedicated functions
+	logger = logger.WithValues("cmd", "top")
+
+	var w = k.Stdout
+	tw := printers.GetNewTabWriter(w)
+	logger.Debug("Tabwriter header created")
+
+	// Build the config from the kubeconfig path
+	config, err := ctrl.GetConfig()
+	if err != nil {
+		return errors.Wrap(err, errKubeConfig)
+	}
+	logger.Debug("Found kubeconfig")
+
+	// Create the clientset for Kubernetes
+	k8sClientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, errCreateK8sClientset)
+	}
+	logger.Debug("Created clientset for Kubernetes")
+
+	// Create the clientset for Metrics
+	metricsClientset, err := versioned.NewForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, errCreateMetricsClientset)
+	}
+	logger.Debug("Created clientset for Metrics")
+
+	ctx := context.Background()
+
+	pods, err := k8sClientset.CoreV1().Pods(c.Namespace).List(ctx, metav1.ListOptions{})
+
+	if err != nil {
+		return errors.Wrap(err, errFetchAllPods)
+	}
+
+	crossplanePods := getCrossplanePods(pods.Items)
+	logger.Debug("Fetched all pods from and filtered out Crossplane", "pods", crossplanePods, "namespace", c.Namespace)
+
+	for i, pod := range crossplanePods {
+		podMetrics, err := metricsClientset.MetricsV1beta1().PodMetricses(pod.PodNamespace).Get(ctx, pod.PodName, metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrap(err, errAddingPodMetrics)
+		}
+		var totalCPUUsage, totalMemoryUsage int64
+		for _, container := range podMetrics.Containers {
+			totalCPUUsage += container.Usage.Cpu().ScaledValue(resource.Milli)
+			totalMemoryUsage += container.Usage.Memory().ScaledValue(resource.Mega)
+		}
+
+		crossplanePods[i].CPUUsage = totalCPUUsage
+		crossplanePods[i].MemoryUsage = fmt.Sprintf("%dMi", totalMemoryUsage)
+	}
+
+	if err != nil {
+		return errors.Wrap(err, errGetPodMetrics)
+	}
+	logger.Debug("Added metrics to Crossplane pods")
+	if c.Split {
+		printPodsByCategory(crossplanePods)
```

Comment:
what's the purpose of the split format? At the moment it contains the exact same information in a less concise format, are you already planning in adding something more down the line?

Reply:
This idea is a result of some team discussions, cc @lsviben . The default output format is the table-like one, however some users might prefer list output, that's why I've added the `--split` flag.

I've been thinking to add some more summary information, for example sum of memory/CPU usage per pod type, count of pods etc.

Reply:
Right, my goal with splitting was that if you have a bunch of functions and providers it would be hard to see/look for what you want if its all mixed up together. Down the line we could add some `--only-functions/providers` flag or a different sort of filtering if it seems useful

Reply:
Got it, I would have expected something like this maybe, the split format just feels dispersive without adding any information:
```
$ kubectl get pods,deployment -A
NAMESPACE            NAME                                                            READY   STATUS    RESTARTS   AGE
crossplane-system    pod/crossplane-7db7b65b8f-jx742                                 1/1     Running   0          5d4h
crossplane-system    pod/crossplane-rbac-manager-864587dcc8-8ddhl                    1/1     Running   0          5d4h
crossplane-system    pod/function-environment-configs-d88b1868aec0-c7b5df9d6-69mqg   1/1     Running   0          5d4h
crossplane-system    pod/function-go-templating-2f9c03dc2482-648d4d58cd-ndzbz        1/1     Running   0          5d3h
crossplane-system    pod/function-patch-and-transform-2472eac62a75-7f844d446-2nm45   1/1     Running   0          5d4h
kube-system          pod/coredns-5d78c9869d-ms9z4                                    1/1     Running   0          5d4h
kube-system          pod/coredns-5d78c9869d-zf8g2                                    1/1     Running   0          5d4h
kube-system          pod/etcd-kind-control-plane                                     1/1     Running   0          5d4h
kube-system          pod/kindnet-tlks6                                               1/1     Running   0          5d4h
kube-system          pod/kube-apiserver-kind-control-plane                           1/1     Running   0          5d4h
kube-system          pod/kube-controller-manager-kind-control-plane                  1/1     Running   0          5d4h
kube-system          pod/kube-proxy-2h9ng                                            1/1     Running   0          5d4h
kube-system          pod/kube-scheduler-kind-control-plane                           1/1     Running   0          5d4h
local-path-storage   pod/local-path-provisioner-6bc4bddd6b-d94ws                     1/1     Running   0          5d4h

NAMESPACE            NAME                                                        READY   UP-TO-DATE   AVAILABLE   AGE
crossplane-system    deployment.apps/crossplane                                  1/1     1            1           5d4h
crossplane-system    deployment.apps/crossplane-rbac-manager                     1/1     1            1           5d4h
crossplane-system    deployment.apps/function-environment-configs-d88b1868aec0   1/1     1            1           5d4h
crossplane-system    deployment.apps/function-go-templating-2f9c03dc2482         1/1     1            1           5d3h
crossplane-system    deployment.apps/function-patch-and-transform-2472eac62a75   1/1     1            1           5d4h
kube-system          deployment.apps/coredns                                     2/2     2            2           5d4h
local-path-storage   deployment.apps/local-path-provisioner                      1/1     1            1           5d4h
```

Reply:
This might be redundant with the default view because:
- we show only pods
- all pods shown are from 1 namespace

Would it be more helpful to show a summary view instead of the current split view?
Something like:
```bash
Nr of pods: 10
Crossplane: 2
Functions: 6
Providers: 2

Resources Utilization:
Memory: 500Mi
CPU: 400
```